docs/agile-development/collaboration/teaming-up.md

```diff
@@ -0,0 +1,65 @@
+# Engagement Team Development in CSE
+
+In every CSE engagement, dynamics are different so are the team requirements. Based on transfer learning among teams, we aim to build right "code-with" environments in every team.
+
+This documentation gives a high-level template with some suggestions by aiming to accelerate team swarming phase to achieve a high speed agility however it has no intention to provide a list of "must-do" items.
+
+## Identification
+
+As it's stated in Tuckman's [team phases](https://en.wikipedia.org/wiki/Tuckman%27s_stages_of_group_development), traditional team development has several stages.
+However those phases can be extremely fast or sometimes mismatched in CSE engagement teams due to external factors.
+
+In order to minimize the risk and set the exceptations on the right way for all parties, an identification phase is important to understand each other.
+Some potential steps in this phase may be as following (not limited):
+
+- [Working agreement](../team-agreements/working-agreements/readme.md)
+
+- Identification of styles/preferences in communication, sharing, learning, decision making of each team member
+  
+  - Talking about necessity of pair programming
+  - Decisions on backlog management & grooming meetings, weekly design sessions, social time sessions...etc.
+  - Sync/Async communication methods, work hours/flexible times
+
+- Decisions and identifications of charts that will be helpful to provide transparent and true information to everyone
+
+- Identification of "Software Craftspersonship" areas which means the tools and methods will be widely used during the engagement and taking the required actions on team upskilling side if necessary.
+
+  - Github, VSCode LiveShare, AzDevOps, necessary development tools & libraries ... more.
+
+- Identification of communication channels, feedback loops and recurrent team call slots out of regular sprint meetings
+
+- Introduction to [Technical Agility Team Manifesto](../team-agreements/team-manifesto/readme.md) and planning the technical delivery by aiming to keep
+technical debt risk minimum.
+
+## Following the Plan and Agile Debugging
```

Comment:
Team needs and configuration may change throughout development. In this case, do you think it is worth add a reminder in the section to review and keep the plan up to date according to the team changes/needs?

Reply:
added below part in 5b86b7bd3759f4ede15c1e9d53c13b0a12cf4867
> Team is also responsible to keep the plan up-to-date based on team changes/needs and debugging results.